pkg/msgconv/from-matrix.go

```diff
@@ -63,12 +64,18 @@ func (mc *MessageConverter) generateContextInfo(ctx context.Context, replyTo *da
 				Msg("Failed to parse reply to message ID")
 		}
 	}
-	if portal.Disappear.Timer > 0 {
-		contextInfo.Expiration = ptr.Ptr(uint32(portal.Disappear.Timer.Seconds()))
-		setAt := portal.Metadata.(*waid.PortalMetadata).DisappearingTimerSetAt
-		if setAt > 0 {
-			contextInfo.EphemeralSettingTimestamp = ptr.Ptr(setAt)
-		}
+	var timer time.Duration
+	if perMessageTimer != nil && perMessageTimer.Timer.Duration > 0 {
+		timer = perMessageTimer.Timer.Duration
+	} else if portal.Disappear.Timer > 0 {
+		timer = portal.Disappear.Timer
+	}
```

Comment:
This should only fall back to the portal timer if `perMessageTimer` is nil, not if `perMessageTimer.Timer.Duration` is 0

```suggestion
	if perMessageTimer != nil {
		timer = perMessageTimer.Timer.Duration
	} else {
		timer = portal.Disappear.Timer
	}
```